sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -136,56 +151,62 @@ public static Read read() {
 
       abstract Builder setInitialPosition(StartingPoint startingPoint);
 
-      abstract Builder setClientProvider(KinesisClientProvider clientProvider);
+      abstract Builder setAWSClientsProvider(AWSClientsProvider clientProvider);
 
       abstract Builder setMaxNumRecords(int maxNumRecords);
 
       abstract Builder setMaxReadTime(Duration maxReadTime);
 
+      abstract Builder setAllowedRecordLateness(Duration allowedRecordLateness);
+
       abstract Read build();
     }
 
     /**
-     * Specify reading from streamName at some initial position.
+     * Specify reading from streamName.
      */
-    public Read from(String streamName, InitialPositionInStream initialPosition) {
-      return toBuilder()
-          .setStreamName(streamName)
-          .setInitialPosition(
-              new StartingPoint(checkNotNull(initialPosition, "initialPosition")))
+    public Read withStreamName(String streamName) {
```

Comment:
Thanks @jkff. I thought there's still time to shape the API of Kinesis connector as it's still in early development stage. It also has the @Experimental(Experimental.Kind.SOURCE_SINK) annotation which states that there might be some incompatible changes in the API.

So in case we want to keep it backward-compatible I would suggest marking old "from" methods as deprecated and keeping the newly added "with" methods (as per your suggestion from https://issues.apache.org/jira/browse/BEAM-1428)

There's was also a rename of KinesisClientProvider interface to AWSClientsProvider. This one fortunately shouldn't break anything as KinesisClientProvider was not public (but it should be as it was fairly pointed out in https://github.com/apache/beam/pull/3540). So the interface along with corresponding builder method "withAWSClientsProvider" can be safely renamed.

What do you think?